### PR TITLE
Simplify Polygon API, now that we always have Rings

### PR DESCRIPTION
**Pull request by dabreegster**
We're almost at the point of changing `Polygon` to just store `Vec<Ring>`. Before making that big rip, I want to fix up some of the methods on `Polygon` and their callers. This PR just does a little of that.

Shouldn't be any behavioral changes, except maybe for some slight performance loss from cloning. That'll go away once the implementation is cutover completely.

### CODE REVIEW

File: geom/src/polygon.rs [hunk 1]
```diff
@@ -161,24 +161,14 @@ impl Polygon {
             &self.points
         }
     }
-    pub fn into_points(mut self) -> Vec<Pt2D> {
```
Comment (dabreegster) [inline geom/src/polygon.rs:164]: Polygon -> points doesn't make sense generally. Callers should ask for the outer ring or the holes or all of them, and do whatever's appropriate

File: geom/src/polygon.rs [hunk 2]
```diff
@@ -161,24 +161,14 @@ impl Polygon {
...
-        if let Some(mut rings) = self.rings.take() {
-            rings.remove(0).into_points()
-        } else {
-            self.points
-        }
-    }
+    // TODO Switch to get_outer_ring. osm2streets depends on this one, have to juggle git repos
```
Comment (dabreegster) [inline geom/src/polygon.rs:164]: Some code now lives in the `osm2streets` repo. There's a circular dependency. I could make the branches point at each other, but it's kind of a hassle. So after this PR, I'm going to push fixes to osm2streets, then update it in this repo, and then proceed with cleanup

File: geom/src/polygon.rs [hunk 3]
```diff
@@ -348,23 +338,28 @@ impl Polygon {
         self.to_geo().intersects(&pl.to_geo())
     }
 
+    // TODO Temporary until we change the internal Polygon representation to always just be rings.
```
Comment (dabreegster) [inline geom/src/polygon.rs:341]: This'll just do `&self.rings` soon

File: geom/src/polygon.rs [hunk 4]
```diff
@@ -348,23 +338,28 @@ impl Polygon {
...
-            Ok(Tessellation::union_all(
-                rings
-                    .iter()
-                    .map(|r| Tessellation::from(r.to_outline(thickness)))
-                    .collect(),
-            ))
-        } else {
-            Ring::new(self.points.clone()).map(|r| Tessellation::from(r.to_outline(thickness)))
-        }
+    pub fn to_outline(&self, thickness: Distance) -> Tessellation {
```
Comment (dabreegster) [inline geom/src/polygon.rs:356]: Most of the PR is just trivially fixing call-sites for this. We can always tessellate an outline for a polygon now!

File: geom/src/polygon.rs (hunk 2 above)
Comment (michaelkirk) [reply geom/src/polygon.rs:164]: reminder that `patches.crates-io` can make this a little less painful, in that at least you can centralize a lot of the changes - but up to you obviously.

File: geom/src/polygon.rs (hunk 4 above)
Comment (michaelkirk) [reply geom/src/polygon.rs:356]: This is a nice cleanup! Though... I don't remember if all the things you are cleaning up were recently introduced as part of all this geometry-juggling, or if it was long standing.

**Review by michaelkirk — APPROVED** [review]

File: geom/src/polygon.rs (hunk 4 above)
Comment (dabreegster) [reply geom/src/polygon.rs:356]: It's always been here, because a `Polygon` (like from SVG) might not've had rings that we can turn into outlines

**Comment by dabreegster** [comment]
Thanks for the review! I'll do the osm2streets bit now, then hopefully that last big step...

**Pull request merged**